Lib/datetime.py

```diff
@@ -1797,13 +1818,13 @@ def utcfromtimestamp(cls, t):
     @classmethod
     def now(cls, tz=None):
         "Construct a datetime from time.time() and optional time zone info."
-        t = _time.time()
+        t = _time.time_ns() / 1e9
```

Comment:
Converting the ns timestamp into a float results in the same precision as `time()` already has.